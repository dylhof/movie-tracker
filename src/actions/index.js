export const storeMovies = (movies) => ({
  type: 'STORE_MOVIES',
  movies
})

export const setCurrentUser = (name, id) => ({
  type: 'SET_USER',
  name,
  id
})

export const logoutCurrentUser = () => ({
  type: 'LOGOUT_USER'
})

export const addFavorite = (id) => ({
  type: 'ADD_FAVORITE',
  id
})

export const addAllUserFavorites = (favorites) => ({
  type: 'ADD_ALL_USER_FAVORITES',
  favorites
})

export const deleteFavorite = (id) => ({
  type: 'DELETE_FAVORITE',
  id
})

export const clearFavorites = () => ({
  type: 'CLEAR_FAVORITES',
})

export const setLoading = (bool) => ({
  type: 'SET_LOADING',
  bool
})