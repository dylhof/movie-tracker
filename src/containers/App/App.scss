body {
  position: relative;
  width: 100%;
  height: 100%;
  // background-image: url('./images/movie-theatre-back.jpg');
  // background-size: cover;
  box-sizing: border-box;
  font-family: 'Righteous', cursive;
}

.App{
  background-image: url('./images/movie-theatre-back.jpg');
  background-attachment: fixed;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  min-height: 950px;
}