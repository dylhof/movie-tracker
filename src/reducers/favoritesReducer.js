export const favoritesReducer = (state = [], action) => {
  switch (action.type) {
    case 'ADD_FAVORITE':
      return [...state, action.id]
    case 'DELETE_FAVORITE':
      return state.filter(id => id !== action.id)
    case 'ADD_ALL_USER_FAVORITES':
      return action.favorites
    case 'CLEAR_FAVORITES':
      return []
    default:
      return state;
  }
}