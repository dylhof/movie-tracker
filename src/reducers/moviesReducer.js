export const moviesReducer = (state = [], action) => {
  switch (action.type) {
    case 'STORE_MOVIES':
      return action.movies
      break;
    default:
      return state;
  }
}