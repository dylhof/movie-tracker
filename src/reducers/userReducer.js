export const userReducer = (state=null, action) => {
  switch (action.type) {
    case 'SET_USER':
      return {name: action.name, id: action.id}
      break;
    default:
      return state;
  }
}