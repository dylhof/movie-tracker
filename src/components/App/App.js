import React, { Component } from 'react';
import '../../main.scss';
import { Route, Switch } from 'react-router-dom'
import  Home  from '../Home/Home'
import { Favorites } from '../Favorites/Favorites'
import { Login } from '../Login/Login'
import { NavBar } from '../NavBar/NavBar'
import {fetchData} from '../../helper/apiCall'
import {getMovies} from '../../actions'
import {connect} from 'react-redux'

class App extends Component {

  fetchAndStoreMovies = async () => {
    const url = 'https://api.themoviedb.org/3/discover/movie?api_key=4340824bb6ffe9ee70c52fc088a91d53&language=en-US&sort_by=release_date.asc&include_adult=true&include_video=false&page=1'
    const moviesData = await fetchData(url)
    const movies = moviesData.results
    this.props.getMovies(movies)
  
  }

  componentDidMount = () => {
 this.fetchAndStoreMovies()

  }

  render() {
    return (
      <div className="App">
        <NavBar />
        <Switch>
          <Route exact path='/' component={Home}></Route>
          <Route exact path='/Favorites' component={Favorites}></Route>
          <Route exact path='/Login' component={Login}></Route>
        </Switch>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  getMovies: (movies) => dispatch(getMovies(movies))
})

export default connect(null, mapDispatchToProps)(App)



// ComponentDidMount 
    // this will check if they have a UN & PW in local storage
        // if yes it will fetch their user info from back end
    // this will make the initial movie fetch
        //then put the movies in store