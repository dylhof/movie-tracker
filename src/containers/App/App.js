import React, { Component } from 'react';
import '../../main.scss';
import { Route, Switch, withRouter, Redirect } from 'react-router-dom'
import Home from '../Home/Home'
import Favorites from '../Favorites/Favorites'
import Login from '../Login/Login'
import SignUp from '../SignUp/SignUp'
import NavBar from '../NavBar/NavBar'
import { fetchData } from '../../helper/apiCall'
import { storeMovies } from '../../actions'
import { connect } from 'react-redux'

export class App extends Component {

  fetchAndStoreMovies = async () => {
    const url = 'https://api.themoviedb.org/3/discover/movie?api_key=4340824bb6ffe9ee70c52fc088a91d53&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1'
    const moviesData = await fetchData(url)
    const movies = moviesData.results
    this.props.dispatchStoreMovies(movies)
  }

  componentDidMount = () => {
    this.fetchAndStoreMovies()
  }

  render() {


    
    return (
      <div className="App">
        <NavBar />
        <Switch>
          <Route exact path='/' component={Home} />
          <Route exact path='/Favorites' render={() => (
            this.props.currentUser ? (
             <Favorites />
            ) : (
                <Redirect to='/Login' />
              )
          )} />

          <Route exact path='/Login' render={() => (
            this.props.currentUser ? (
              <Redirect to='/' />
            ) : (
                <Login />
              )
          )} />
          <Route exact path='/SignUp' render={() => (
            this.props.currentUser ? (
              <Redirect to='/' />
            ) : (
                <SignUp />
              )
          )} />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  currentUser: state.currentUser
})

export const mapDispatchToProps = (dispatch) => ({
  dispatchStoreMovies: (movies) => dispatch(storeMovies(movies))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App))



// ComponentDidMount 
    // this will check if they have a UN & PW in local storage
        // if yes it will fetch their user info from back end
    // this will make the initial movie fetch
        //then put the movies in store