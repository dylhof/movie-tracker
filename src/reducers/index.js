import { combineReducers } from 'redux';
import {moviesReducer} from './moviesReducer';
import {userReducer} from './userReducer';
import {favoritesReducer} from './favoritesReducer';
import {loadingReducer} from './loadingReducer';
import {errorReducer} from './errorReducer';

export const rootReducer = combineReducers({
  movies: moviesReducer,
  currentUser: userReducer,
  favorites: favoritesReducer,
  isLoading: loadingReducer,
  error: errorReducer,
}) 