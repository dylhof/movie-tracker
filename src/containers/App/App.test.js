import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme'
import { mapDispatchToProps } from './App'
import * as actions from '../../actions'
import { Provider } from 'react-redux'
import { createStore } from 'redux';
import { rootReducer } from '../../reducers'
import { BrowserRouter } from 'react-router-dom';

describe('App', () => {
  const store = createStore(rootReducer)
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // it('should return an object with a movies array', () => {
  //   //setup
  //   const mockState = {
  //     movies: [{ title: 'Aquaman' }],
  //     favorites: [{}]
  //   }
  //   const expected = {
  //     movies: [{ title: 'Aquaman'}]
  //   }
  //   //execution
  //   const mappedProps = mapState
  // })
  describe('mapDispatchToProps', () => {

    it('should call dispatch when using a function from MDTP', () => {
      //setup
      const mockDispatch = jest.fn()
      const actionToDispatch = actions.storeMovies([{}])
      //execution
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.dispatchStoreMovies([{}])
      //expectation
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})