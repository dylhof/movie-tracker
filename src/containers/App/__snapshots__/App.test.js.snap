// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot if there is a current user 1`] = `
<div
  className="App"
>
  <Connect(NavBar) />
  <div
    className="error"
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      path="/favorites"
      render={[Function]}
    />
    <Route
      path="/login"
      render={[Function]}
    />
    <Route
      path="/signUp"
      render={[Function]}
    />
    <Route
      path="/movie/:id"
      render={[Function]}
    />
  </Switch>
</div>
`;

exports[`App should match the snapshot if there is not a current user 1`] = `
<div
  className="App"
>
  <Connect(NavBar) />
  <div
    className="error"
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      path="/favorites"
      render={[Function]}
    />
    <Route
      path="/login"
      render={[Function]}
    />
    <Route
      path="/signUp"
      render={[Function]}
    />
    <Route
      path="/movie/:id"
      render={[Function]}
    />
  </Switch>
</div>
`;
